android/app/src/main/java/com/microsoft/reactnativecodepush/CodePush.java

```diff
@@ -0,0 +1,412 @@
+                        && packageHash != null
+                        && packageHash.length() > 0
+                        && packageHash.equals(codePushPackage.getCurrentPackageHash());
+                resolve.invoke(isFirstRun);
+            } catch (IOException e) {
+                e.printStackTrace();
+                reject.invoke(e.getMessage());
+            }
+        }
+
+        @ReactMethod
+        public void notifyApplicationReady(Callback resolve, Callback reject) {
+            cancelRollbackTimer();
+            resolve.invoke("");
+        }
+
+        @ReactMethod
+        public void setUsingTestFolder(boolean shouldUseTestFolder) {
+            usingTestFolder = shouldUseTestFolder;
+        }
+
+        @ReactMethod
+        public void restartImmediateUpdate(int rollbackTimeout) {
+            initializeUpdateWithRollbackTimeout(rollbackTimeout, /*needsRestart*/ true);
+        }
+
+        @ReactMethod
+        public void restartPendingUpdate() {
+            checkForPendingUpdate(/*needsRestart*/ true);
+        }
+
+        @Override
+        public Map<String, Object> getConstants() {
+            final Map<String, Object> constants = new HashMap<>();
+            constants.put("codePushInstallModeImmediate", CodePushInstallMode.IMMEDIATE.getValue());
+            constants.put("codePushInstallModeOnNextRestart", CodePushInstallMode.ON_NEXT_RESTART.getValue());
+            constants.put("codePushInstallModeOnNextResume", CodePushInstallMode.ON_NEXT_RESUME.getValue());
+            return constants;
+        }
+
+        public CodePushNativeModule(ReactApplicationContext reactContext) {
+            super(reactContext);
+        }
+
+        @Override
+        public String getName() {
+            return "CodePush";
+        }
+    }
+
+    private class CodePushReactPackage implements ReactPackage {
+        @Override
+        public List<NativeModule> createNativeModules(ReactApplicationContext reactApplicationContext) {
+            List<NativeModule> nativeModules = new ArrayList<>();
+            CodePush.this.codePushNativeModule = new CodePushNativeModule(reactApplicationContext);
+            CodePushDialog dialogModule = new CodePushDialog(reactApplicationContext, mainActivity);
+
+            nativeModules.add(CodePush.this.codePushNativeModule);
+            nativeModules.add(dialogModule);
+
+            reactApplicationContext.addLifecycleEventListener(new LifecycleEventListener() {
+                @Override
+                public void onHostResume() {
+                    CodePush.this.checkForPendingUpdateDuringResume();
+                }
+
+                @Override
+                public void onHostPause() {
+                }
+
+                @Override
+                public void onHostDestroy() {
+                }
+            });
+
+            return nativeModules;
+        }
+
+        @Override
+        public List<Class<? extends JavaScriptModule>> createJSModules() {
```

Comment:
We are implementing an interface, so if you do not override the method, the compiler will throw an error.